### PR TITLE
new flag  --namespace

### PR DESCRIPTION
**Pull request by prezha**
fixes: #2117 

add support for namespace:
- as `--namespace` flag for `start` command
- in minikube config
- in kubeconfig

### example

~/.kube/config:
```
apiVersion: v1
clusters:
contexts:
current-context: ""
kind: Config
preferences: {}
users:
```
❯ minikube start --namespace='production'
```
😄  minikube v1.14.0 on "Opensuse-Tumbleweed"
✨  Automatically selected the docker driver
❗  docker is currently using the btrfs storage driver, consider switching to overlay2 for better performance
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=16000MB) ...
🐳  Preparing Kubernetes v1.19.2 on Docker 19.03.8 ...
    > kubeadm.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
    > kubelet.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
    > kubectl.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
    > kubectl: 41.01 MiB / 41.01 MiB [----------------] 100.00% 7.12 MiB p/s 6s
    > kubelet: 104.88 MiB / 104.88 MiB [-------------] 100.00% 6.70 MiB p/s 16s
    > kubeadm: 37.30 MiB / 37.30 MiB [---------------] 100.00% 2.12 MiB p/s 17s
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "production" namespace by default
```
~/.minikube/profiles/minikube/config.json:
```
{
  "Name": "minikube",
...
  "KubernetesConfig": {
    "KubernetesVersion": "v1.19.2",
    "ClusterName": "minikube",
    "Namespace": "production",
...
  },
...
}
```
~/.kube/config:
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority: /home/prezha/.minikube/ca.crt
    server: https://192.168.49.2:8443
  name: minikube
contexts:
- context:
    cluster: minikube
    namespace: production
    user: minikube
  name: minikube
current-context: minikube
kind: Config
preferences: {}
users:
- name: minikube
  user:
    client-certificate: /home/prezha/.minikube/profiles/minikube/client.crt
    client-key: /home/prezha/.minikube/profiles/minikube/client.key
```

### CODE REVIEW

File: cmd/minikube/cmd/start_flags.go [hunk 1]
```diff
@@ -153,6 +154,7 @@ func initMinikubeFlags() {
 // initKubernetesFlags inits the commandline flags for Kubernetes related options
 func initKubernetesFlags() {
 	startCmd.Flags().String(kubernetesVersion, "", fmt.Sprintf("The Kubernetes version that the minikube VM will use (ex: v1.2.3, 'stable' for %s, 'latest' for %s). Defaults to 'stable'.", constants.DefaultKubernetesVersion, constants.NewestKubernetesVersion))
+	startCmd.Flags().String(startNamespace, "default", "The namespace used in the Kubernetes cluster")
```
Comment (medyagh) [inline cmd/minikube/cmd/start_flags.go:157]: a better description would be, named space to activate after start.

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by prezha** [comment]
it took me some time to determine why travis tests have been failing with:
```
--- FAIL: TestGenerateDocs (0.02s)
    --- FAIL: TestGenerateDocs/pause (0.00s)
        generate-docs_test.go:47: Docs are not updated. Please run `make generate-docs` to update commands documentation:   strings.Join({
              	... // 9 identical lines
              	"pause Kubernetes",
              	"",
            + 	"### Synopsis",
            + 	"",
            + 	"pause Kubernetes",
            + 	"",
              	"```",
              	"minikube pause [flags]",
              	... // 30 identical lines
              }, "\n")
    --- FAIL: TestGenerateDocs/unpause (0.01s)
        generate-docs_test.go:47: Docs are not updated. Please run `make generate-docs` to update commands documentation:   strings.Join({
              	... // 9 identical lines
              	"unpause Kubernetes",
              	"",
            + 	"### Synopsis",
            + 	"",
            + 	"unpause Kubernetes",
            + 	"",
              	"```",
              	"minikube unpause [flags]",
              	... // 30 identical lines
              }, "\n")
```
as `pause`/`unpause` should not be affected by changes in this pr, then i noticed that `go.mod` somehow got updated, upgrading also `spf13/cobra` from `v1.0.0` to `v1.1.1`, which could be the 'culprit', so i've reverted back `go.mod` and it worked

bottom line: **upgrade of `spf13/cobra` from `v1.0.0` to `v1.1.1` could potentialy cause some strange side-effects & issues**

File: cmd/minikube/cmd/start_flags.go (hunk 1 above)
Comment (prezha) [reply cmd/minikube/cmd/start_flags.go]: i've changed the description accordingly

**Review by medyagh — APPROVED** [review]
thank you for this PR
